src/mvt_getter.py

```diff
@@ -0,0 +1,383 @@
+                    (select ST_Union(geom) as geom from
+                      (select ST_Buffer(geom, %s, 0) geom from
+                         water_polygons_vector
+                         where
+                            geom && ST_TileEnvelope(%s, %s, %s)
+                      ) p
+                    ) p
+                    where ST_Area(geom) > %s
+                  ) p
+          """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            groupby,
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            groupby,
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+            (pixel_size_at_zoom(maxzoom, pxtolerance) ** 2) / pxtolerance,
+            groupby,
+            pixel_size_at_zoom(maxzoom, pxtolerance) ** 2,
+            minzoom,
+            x,
+            y,
+            ",".join([(("'coastline'" if name == 'natural' else 'null') + ' as "' + name + '"') for name in column_names]),
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            minzoom,
+            x,
+            y,
+            pixel_size_at_zoom(maxzoom, pxtolerance) ** 2
+        )
+    elif vec == "line":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s from
+                        (select ST_LineMerge(way) as %s, %s from
+                            (select ST_Union(way) as %s, %s from
+                                %s
+                                where (%s)
+                                and way && ST_TileEnvelope(%s, %s, %s)
+                            group by %s
+                            ) p
+                        ) p
+          """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+            ",".join([escape_sql_column(name, type=types[vec], asname=False) for name in column_names]),
+        )
+    elif vec == "point":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s
+                        from %s where (%s)
+                        and way && ST_TileEnvelope(%s, %s, %s)
+                        limit 10000
+                 """ % (
```

Comment:
[https://github.com/kothic/kothic/pull/26/files#r656830579](@Komzpa ):

> slightly better idea is to define order by admin_level desc desc nulls last, population desc nulls last and do a select distinct on (ST_SnapToGrid(way, 5px))

Reply:
hm, I don't think that doing `ST_SnapToGrid(way, 5px)` is needed since having multiple points with near the same coordinates seems like valid case. ordering is a good idea, currently there is sorting by population that happens on client side, but doing it in db seems better.